pkg/cloudprovider/providers/azure/azure_blobDiskController.go

```diff
@@ -217,6 +215,8 @@ func (c *BlobDiskController) deleteVhdBlob(accountName, accountKey, blobName str
 func (c *BlobDiskController) CreateBlobDisk(dataDiskName string, storageAccountType storage.SkuName, sizeGB int) (string, error) {
 	klog.V(4).Infof("azureDisk - creating blob data disk named:%s on StorageAccountType:%s", dataDiskName, storageAccountType)
 
+	c.initStorageAccounts()
```

Comment:
I haven't looked at the other BlobDisk operations (attach/detach/delete) but I'm assuming here that they cannot get called unless `CreateBlobDisk` was called

Reply:
yes, only `CreateBlobDisk` func could invoke `getAllStorageAccounts`